packages/geo/src/lib/layer/shared/layers/vector-layer.ts

```diff
@@ -452,12 +460,29 @@ export class VectorLayer extends Layer {
         ? new olProjection({ code: paramsWFS.srsName })
         : proj;
       const currentExtent = olproj.transformExtent(extent, proj, wfsProj);
+      const ogcFilters = (options as OgcFilterableDataSourceOptions).ogcFilters;
+
+      if (
+        (this.previousLoadExtent &&
+          this.previousLoadExtent !== currentExtent) ||
+        (this.previousLoadResolution &&
+          this.previousLoadResolution !== resolution) ||
+        (this.previousOgcFilters && this.previousOgcFilters !== ogcFilters)
+      ) {
+        vectorSource.removeLoadedExtent(this.previousLoadExtent);
+        this.xhrAccumulator.map((xhr) => xhr.abort());
```

Comment:
We could use the forEach instead of the map, it's more efficient for a simple iteration and even better would be the "for of". I think it's important in the scenario that this could be a very in-demand method.

Reply:
done